common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -543,6 +543,9 @@ public UTF8String repeat(int times) {
    * @return the position of the first occurrence of substr, if not found, -1 returned.
    */
   public int indexOf(UTF8String v, int start) {
+    if (start < 0) {
+      return -1;
```

Comment:
if we're trying to follow Hive behavior for locate then we should move this logic to `StringLocate` instead